mu/harness/board_sim.py

```diff
@@ -0,0 +1,68 @@
+import threading
+
+from mu.harness import project
+from mu.harness import decoder
+from mu.protogen import stores_pb2
+from mu.protogen import gpio_pb2
+
+from mu.harness.project import StoreUpdate
+
+# Constants for stm32f0xx store keys
+GPIO_KEY = (stores_pb2.MuStoreType.GPIO, 0)
+
+class BoardSim:
+    def __init__(self, pm, proj_name, sub_sim_classes=None, init_conds=None):
+        self.pm = pm
+        self.sub_sims = {}
+        if sub_sim_classes:
+            for sub_sim_class in sub_sim_classes:
+                self.sub_sims[sub_sim_class.__name__] = sub_sim_class(self)
```

Comment:
I think that because there's not that many cases of having multiple similar boards, I'm ok with just forcing writing multiple sims. If that changes we can always refactor.

I implemented the `sub_sim` function but normalized to lower case not upper case.